### PR TITLE
[LLM]Add generate_stream in autoTP serving Example

### PR DESCRIPTION
**Pull request by Romanticoseu**
## Description

Add `generate_stream()` API for `DeepSpeed AutoTp FastApi` example

### 2. User API changes

Implement a class called `TextIteratorStreamer`, adapted from transformers `TextIteratorStreamer`.

### 4. How to test?
- [ ] Application test

### CODE REVIEW

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py [hunk 1]
```diff
@@ -193,19 +285,48 @@ async def process_requests():
...
+
+            # import pdb
+            # pdb.set_trace()
+
+            await generate_stream_gate(
+                [req.prompt for req in object_list],
+                [req.n_predict for req in object_list],
+                request_ids,
+            )
+            # for request_id in request_ids:
```
Comment (xiangyuT) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py:305]: Remove unused comments

Comment (Romanticoseu) [reply python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py:305]: I will change this later.

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py [hunk 2]
```diff
@@ -155,6 +214,35 @@ async def generate(prompt_request: PromptRequest):
...
+                while i < prompt_request.n_predict:
+                    if not token_queue.empty():
+                        token = await token_queue.get()
+                        logger.error(f"Consumer: {token}")
+                        # yield token
+                        yield json.dumps(
+                            {"request_id": request_id, "text": token}
+                        ) + "\n"
+                    else:
+                        await asyncio.sleep(0.001)
```
Comment (xiangyuT) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py:236]: Is `0.001` a magic setting? Can we align `asyncio.sleep()` to  `asyncio.sleep(0)` or `asyncio.sleep(0.1)`?

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py [hunk 3]
```diff
@@ -131,19 +139,70 @@ def generate_text(prompt: List[str], n_predict = 32):
...
+        torch.xpu.synchronize()
+
+    t1 = Thread(target=model_generate)
+    t1.start()
+
+    partial_output = ""
+    rfind_start = 0
+
+    async def put_item(queue, item):
+        logger.error(f"Producer: {item}")
```
Comment (xiangyuT) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py:183]: Remove logger.error

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py [hunk 4]
```diff
@@ -89,70 +115,144 @@ def load_model(model_path, low_bit):
...
+            return {"generated_text": "".join(output_str)}
+
+
+@app.post("/generate_stream/")
+async def generate_stream(prompt_request: PromptRequest):
+    request_id = str(uuid.uuid4()) + "stream"
+    await request_queue.put((request_id, prompt_request))
+    while True:
+        await asyncio.sleep(0.1)
+        if request_id in streamer_dict and request_id.endswith("stream"):
```
Comment (xiangyuT) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py:249]: Is the check ` request_id.endswith("stream")` necessary? `request_id` should be init in line 245 and not change in the function?

**Review by xiangyuT — APPROVED** [review]
LGTM

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request closed without merging**